lib/ui/page/buy.dart

```diff
@@ -0,0 +1,437 @@
+                    ],
+                    keyboardType:
+                        const TextInputType.numberWithOptions(decimal: true),
+                    textAlign: TextAlign.end,
+                  ),
+                ),
+                const SizedBox(width: 20),
+              ]),
+            ),
+            const SizedBox(height: 20),
+            RoundContainer(
+              height: 64,
+              padding: const EdgeInsets.symmetric(vertical: 0, horizontal: 10),
+              child: Row(children: [
+                InkWell(
+                    onTap: showAssetListBottomSheet,
+                    child: SymbolIconWithBorder(
+                      symbolUrl: asset.value.iconUrl,
+                      chainUrl: asset.value.chainIconUrl,
+                      size: 40,
+                      chainBorder:
+                          const BorderSide(color: Color(0xfff8f8f8), width: 1),
+                      symbolBorder:
+                          const BorderSide(color: Color(0xfff8f8f8), width: 2),
+                      chainSize: 8,
+                    )),
+                const SizedBox(width: 10),
+                InkWell(
+                    onTap: showAssetListBottomSheet,
+                    child: Text(
+                      asset.value.symbol,
+                      style: TextStyle(
+                        color: context.colorScheme.primaryText,
+                        fontSize: 14,
+                        fontWeight: FontWeight.w400,
+                      ),
+                    )),
+                const SizedBox(width: 10),
+                SvgPicture.asset(R.resourcesIcArrowDownSvg),
+                const SizedBox(width: 10),
+                Expanded(
+                    child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.end,
+                  children: [
+                    SizedBox(
+                        height: 38,
+                        child: TextField(
+                          focusNode: cryptoFocusNode,
+                          style: TextStyle(
+                            color: context.colorScheme.primaryText,
+                            fontSize: 16,
+                            fontWeight: FontWeight.w400,
+                          ),
+                          maxLines: 1,
+                          decoration: const InputDecoration(
+                            hintText: '0.000',
+                            border: InputBorder.none,
+                          ),
+                          controller: cryptoController,
+                          inputFormatters: <TextInputFormatter>[
+                            FilteringTextInputFormatter.allow(
+                                RegExp(r'^\d*\.?\d{0,3}'))
+                          ],
+                          keyboardType: const TextInputType.numberWithOptions(
+                              decimal: true),
+                          textAlign: TextAlign.end,
+                        )),
```

Comment:
the crypto textfield has the same style as fait textfield ?  what about extract them to another widget